types/wlr_screencopy_v1.c

```diff
@@ -269,18 +293,39 @@ static void frame_handle_copy(struct wl_client *wl_client,
 		return;
 	}
 
-	struct wl_shm_buffer *buffer = wl_shm_buffer_get(buffer_resource);
-	if (buffer == NULL) {
+	struct wlr_dmabuf_v1_buffer *dma_buffer = NULL;
+	struct wl_shm_buffer *shm_buffer = wl_shm_buffer_get(buffer_resource);
+	if (shm_buffer == NULL) {
+		dma_buffer =
+			wlr_dmabuf_v1_buffer_from_buffer_resource(buffer_resource);
```

Comment:
This function aborts if the resource isn't a dmabuf. We need to use `wlr_dmabuf_v1_resource_is_buffer` to check before calling it.